### PR TITLE
creating models for the db

### PR DESCRIPTION
**Pull request by as005415**

### CODE REVIEW

File: Storage/Estate.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿namespace Storage
+{
+    public class Estate
+    {
+        public int Id { get; set; }
+        public int CommonArea { get; set; }
```
Comment (siarheivlasko) [inline Storage/Estate.cs:6]: make them double

File: Storage/Estate.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        public int LivingArea { get; set; }
+        public string Address { get; set; }
+        public string TypeOfProperty { get; set; }
+        public string StateOfProperty { get; set; }
+        
+        public int PersonRequestId { get; set; }
```
Comment (siarheivlasko) [inline Storage/Estate.cs:12]: I think that relation is redundant, request -> person -> estate

File: Storage/StorageContext.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            modelBuilder.Entity<Estate>()
+                .HasOne<PersonRequests>(info => info.PersonRequest)
+                .WithMany(requests => requests.Estates)
+                .HasForeignKey(info => info.PersonRequestId);
+            
+
+            modelBuilder.Entity<UsersPersonRequests>().HasKey(r => new {r.UserId, r.PersonRequestId});
+            modelBuilder.Entity<UsersPersonRequests>().HasIndex(r => new {r.UserId, r.PersonRequestId}).IsUnique();
+            
+            modelBuilder.Entity<UsersPersonRequests>()
```
Comment (siarheivlasko) [inline Storage/StorageContext.cs:28]: Use separate configuration for each model https://www.entityframeworktutorial.net/code-first/move-configurations-to-seperate-class-in-code-first.aspx

File: Storage/StorageContext.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Estate>()
+                .HasOne<PersonRequests>(info => info.PersonRequest)
+                .WithMany(requests => requests.Estates)
+                .HasForeignKey(info => info.PersonRequestId);
+            
+
+            modelBuilder.Entity<UsersPersonRequests>().HasKey(r => new {r.UserId, r.PersonRequestId});
```
Comment (siarheivlasko) [inline Storage/StorageContext.cs:25]: do we really need this keY?

File: Storage/StorageContext.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                .WithMany(usersPersonRequest => usersPersonRequest.UsersPersonRequests)
+                .HasForeignKey(requests => requests.UserId);
+            
+            modelBuilder.Entity<UsersPersonRequests>()
+                .HasOne<PersonRequests>(requests => requests.PersonRequest)
+                .WithMany(usersPersonRequest => usersPersonRequest.UsersPersonRequests)
+                .HasForeignKey(requests => requests.PersonRequestId);
+
+
+            modelBuilder.Entity<UsersRoles>().HasKey(r => new {r.UserId, r.RoleId});
```
Comment (siarheivlasko) [inline Storage/StorageContext.cs:39]: Do we really need this key?

**Review by siarheivlasko — COMMENTED** [review]
Not bad!

File: Storage/Models/Queue.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Storage.Models
+{
+    public class Queue
+    {
+        public int Id { get; set; }
+        public int Number { get; set; }
```
Comment (siarheivlasko) [inline Storage/Models/Queue.cs:6]: Let us use Date of request for numbering in queue

**Review by siarheivlasko — APPROVED** [review]

File: Storage/Models/Queue.cs (hunk 6 above)
Comment (as005415) [reply Storage/Models/Queue.cs:6]: We can access Date of request through PersonRequestId

**Pull request merged**